### PR TITLE
Added cluster tool link in cluster group nav section

### LINKED ISSUE #9228: Relocate Cluster Tools button
(Design for Q3, implement for Q4)

In 2.6, with the new UI we added a Cluster Tools button to the bottom of the side nav when exploring a cluster, so users could easily get to the tools section - to help with migrating from the old UI.

image

Now we're at 2.7, we should figure out where to put cluster tools so that we can remove this item from the side bar.

Suggestion would be a `Tools` item under the Cluster nav, i.e.

image

I'd also suggest adding a link to the Cluster Dashboard page:

image

Suggestion would be to stack the links 'Add cluster badge and install monitoring vertically and to add 'Manage Cluster Tools' in there as well - e.g.

image

We could take the opportunity to make that header look more visual - we should show the infrastructure provider as well as the k8s provider.

### PR DESCRIPTION
**Pull request by bisht-richa**
### Summary
Fixes #9228

Relocate the Cluster Tools button

Note: The `ifHaveType` mechanism did not support multiple types - this has been added in this PR, so that we show the Tools item in the sidenav only when user has access to the required types.

### How to test
Ref XD link: https://xd.adobe.com/view/30c5a51c-e7b7-4ebf-b75f-1dc37391d3d7-6596/

- Go to local cluster => side nav should have `Tools` nav link under `Cluster` group
- `Cluster Tools` Should not be there at the bottom of side nav
-  On the cluster dashboard page > `Install Monitoring` text has changed to `Cluster Tools` and 
`Add Cluster Badge` link has changed to an icon button with a tooltip 'Customize Appearance'

### Manual Tests

- Check that the side nav renders correctly
- Check the icon button opens the customize appearance dialog
- Check that clicking on 'Tools' in the side nav takes you to cluster tools
- Check that clicking on 'Cluster Tools' on the dashboard page takes you to the cluster tools page
- Install a cluster tool like monitoring and ensure that the side nav updates to show 'Monitoring'

### Screenshot/Video

image

### CODE REVIEW

File: shell/pages/c/_cluster/explorer/index.vue [hunk 1]
```diff
@@ -177,6 +177,11 @@ export default {
       return this.$store.getters['management/all'](MANAGEMENT.CLUSTER);
     },
 
+    showClusterTools() {
```
Comment (richard-cox) [inline shell/pages/c/_cluster/explorer/index.vue:180]: These permissions need also to be applied to the tools virtual type in the explorer config.  Looks like `ifHaveType` can be a regex string, so multiple types should work. 

Notice we have `ifHaveType:          new RegExp(`${ MANAGEMENT.SETTING }|${ MANAGEMENT.FEATURE }`, 'i'),` elsewhere... which i don't think should actually work given below? Maybe `RegExp` ctor has smarts to accept regex objs?
- `const haveIds = knownTypes[module].filter((t) => t.match(stringToRegex(p.ifHaveType)) );`?
- ```
    function stringToRegex(str) {
      let out = regexCache[str];
    
      if ( !out ) {
        out = new RegExp(str);
        regexCache[str] = out;
      }
    
      return out;
     }
   ```

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: shell/pages/c/_cluster/explorer/index.vue (hunk 1 above)
Comment (nwmac) [reply shell/pages/c/_cluster/explorer/index.vue:187]: Done

**Review by aalves08 — APPROVED** [review]
testing and codewise LGTM

Screenshot 2024-05-23 at 15 25 57

**Review by nwmac — APPROVED** [review]

**Comment by nwmac** [comment]
@richard-cox @aalves08 has approved - need your approval to review since you had requested changes.

**Review by richard-cox — APPROVED** [review]

**Pull request merged**